__e2e__/root.test.ts

```diff
@@ -53,6 +53,15 @@ test('works when Gradle is run outside of the project hierarchy', () => {
    */
   const gradleWrapper = path.join(androidProjectRoot, 'gradlew');
 
+  // Make sure that we use `-bin` distribution of Gradle
+  await spawnScript(
+    gradleWrapper,
+    ['wrapper', '--gradle-version=8.0.1', '--distribution-type', 'bin'],
+    {
+      cwd: androidProjectRoot,
+    },
+  );
```

Comment:
I don't think this is necessary 🤔 I've been using gradle/gradle-build-action with several projects which have the `-all` distribution and I haven't noticed re-downloads of Gradle.

Reply:
So I gave it a try and it is getting re-downloaded :( See [here](https://github.com/react-native-community/cli/actions/runs/7037142009/job/19151295313?pr=2150). 

![CleanShot 2023-11-29 at 19 53 33@2x](https://github.com/react-native-community/cli/assets/63900941/2129e568-51e5-4e4e-9cf7-783b9049ba4a)

Reply:
Great thanks for clarifying